lib/matplotlib/path.py

```diff
@@ -558,14 +558,12 @@ def intersects_bbox(self, bbox, filled=True):
         :class:`~matplotlib.transforms.Bbox`.
 
         *filled*, when True, treats the path as if it was filled.
-        That is, if one path completely encloses the other,
-        :meth:`intersects_path` will return True.
+        That is, if the path completely encloses the bounding box,
+        :meth:`intersects_bbox` will return True.
+
+        The bounding box is always considered filled.
```

Comment:
The name 'bounding box' already implies that the box represents the boundary of some more complex object which is somewhere inside the bounding box, so I think the new behavior makes more sense.

In order to retain the old behavior, it would be necessary to add an additional option, because we would actually need three use cases:

- Filled path intersects filled bbox (filled=True, unchanged, needed in many places).
- Unfilled path intersects unfilled bbox (filled=False, old behavior).
- Unfilled path intersects filled bbox (filled=False, new behavior, needed in legend).

So it looks like that would *also* result in an API change, unless the ```filled``` argument accepts more than two possible values.

If you agree, I will keep the behavior as I've implemented it now and update ```api_changes```.